### ISSUE TITLE
Implement REST API authentication tokens

### ISSUE THREAD

**Issue by lucaderi** [ntop/ntopng#4120]
In ntopng authentication is based on username+password. This ticket is to request the implementation of authentication tokens used in the REST API instead of username+password. This means that REST API users can ether provide
- username+password
or
- token (without username+password)

in REST API calls.

Tokens are bound to users (ie. a user has up to one token). The token is used ONLY for the REST API, namely NON-REST scripts (e.g. the GUI) will ignore the ticket for authentication.

Tokens can be created/revoked in the user interface prior to use them in the REST API.

See also https://v2.docs.influxdata.com/v2.0/security/tokens/

**Comment by TimotheusBachinger** [comment]
Hi Luca,

thanks for the proposal.

We would have a follow-up question regarding the implementation.

- Imagine a situation where multiple checkmk users have different ntop accounts
- In order to get a token for every checkmk user's ntop account, all credentials of those ntop accounts must be stored on the checkmk server OR all the password must be queried (e.g. via a pop-up)
- Or could we only store a ntop admin's account credentials and retrive a cookie for a specific ntop user by passing the user name as a parameter? I am posting some python example code to illustrate our idea:
```python

import requests
url4 = "http://ntop4.lan.tribe29.com:3000/authorize.html"
mySession = requests.Session()
form_data = {"user": "admin_user", "password": "*****", "user_to_get_cookie_for":"non-priviledged-ntop-user"}
r = mySession.post(url4, form_data)
mySession.cookies.get_dict() # --> will contain cookie for user "non-priviledged-ntop-user"
r = mySession.get(<any ntop REST GET call>) # will return data as "non-priviledged-ntop-user" is allowed to see
```
Regards
Timi

**Comment by TimotheusBachinger** [comment]
Hi Luca,

as discussed, the implementation as in [https://github.com/ntop/ntopng/issues/4220](https://github.com/ntop/ntopng/issues/4220 ) would be suitable.
Furthermore the session token will not expire, so there is no need to refresh it from our side.

You could close this issue, as it seems to be already implemented with #4220 

Regards
Timi

**Comment by lucaderi** [comment]
Closed as requested but in #4220 the cookie expires and it's not a token as proposed.
